### ISSUE TITLE
emacsclient arguments overridden

### ISSUE THREAD

**Issue by nomennescio** [factor/factor#2065]
See #1808 

The new code for emacsclient uses a custom argument variable, but it immediately is overridden

```
M: emacsclient editor-command
    [
        emacsclient-path get [ find-emacsclient ] unless* ,
        emacsclient-args get % "-a=emacs" , "--no-wait" ,
        number>string "+" prepend , ,
    ] { } make ;
```

In this case I have a different path to EDITOR (the value passed to '-a'). If I supply my own '-a' argument in emacsclient-args, the emacsclient executable will use the LAST argument, and overrides my argument.

It's a design choice if you want to force these arguments even if emacsclient-args is non-empty, but in that case allow the user at least to be able to override the supplied arguments. I therefore propose to change the code into

```
M: emacsclient editor-command
    [
        emacsclient-path get [ find-emacsclient ] unless* ,
        "-a=emacs" , "--no-wait" , emacsclient-args get %
        number>string "+" prepend , ,
    ] { } make ;
```

**Comment by mrjbq7** [comment]
Do you think we should default emacsclient-args to the current arguments (-a —no-wait) or just add it at the end to allow overriding?

**Comment by nomennescio** [comment]
I think having default arguments is the better design.

**Comment by mrjbq7** [comment]
Fixed in a51fd04d0b88bdbe16fc6adbc5fe2e0b9a68fa66.
